### DISCUSSION TITLE
Generating commission based on Paid amount on invoices

### DISCUSSION THREAD

**Discussion by Krosslaut** [OCA/odoo-community.org#153] [Q&A] [answered]
Hi everyone, i've been having troubles using the commission formula to get commissions based on paids made on invoices, is there any way to do that? thanks in advance for the help

**Comment by pedrobaeza** [comment]
Well, I meant in https://github.com/OCA/commission/discussions, but no problem having it here.

For getting that, just select this option:

![image](https://github.com/OCA/odoo-community.org/assets/7165771/0fceb3a7-17f6-42e0-857e-e52adf42a51f)

**Reply by pedrobaeza** [reply]
This option doesn't allow partial payments though.

**Reply by Krosslaut** [reply]
Ups, thanks i'll check that but if i do it that way, i'll want to remove the tax from the commissions, that's why i was trying to use the formulas, but those seems to work only with invoice lines, not with paid amounts.

**Reply by Krosslaut** [reply]
Yep i see that now, i'm having a hard time trying to generate commission based on payments instead but the logic changes completely

**Reply by jguenat** [reply]
As Pedro said this option will only let you settle the commission when the invoice is fully paid.
I don't think you can manage partial payment without modifying the code.

As for removing the tax from the commission, I did it with the following formula ( 20% commission based on unit price tax excluded) 
```
if line._name == 'sale.order.line':
    result = 0
if line._name == 'account.move.line':
    if line.tax_ids and line.tax_ids[0].price_include:
        result = (line.price_unit / (100 + line.tax_ids[0].amount) * 100) * 0.20 * line.quantity
    else:
        result = line.price_unit * 0.20 * line.quantity
```
(it obviously does not support multiple taxes)

**Reply by Krosslaut** [reply]
Ah nice solution for the tax side, and yeah, i'll have to made some test, if i find something interesting i'll post it here

**Comment by francesco-ooops** [comment, marked as answer]
@Krosslaut could this be what you're looking for? https://github.com/OCA/commission/tree/14.0/sale_commission_partial_settlement

**Reply by Krosslaut** [reply]
Yep seems like this one is exactly what i need, i'll check if it works right away or i have to port something, Thanks a lot, i didn't check previous versions of that module
